extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/MSQTaskQueryMaker.java

```diff
@@ -140,23 +150,13 @@ public QueryResponse<Object[]> runQuery(final DruidQuery druidQuery)
 
     // This parameter is used internally for the number of worker tasks only, so we subtract 1
     final int maxNumWorkers = maxNumTasks - 1;
-
-    final int rowsPerSegment = MultiStageQueryContext.getRowsPerSegment(
-        queryContext,
-        DEFAULT_ROWS_PER_SEGMENT
-    );
-
-    final int maxRowsInMemory = MultiStageQueryContext.getRowsInMemory(
-        queryContext,
-        DEFAULT_ROWS_IN_MEMORY
-    );
-
-    final IndexSpec indexSpec = MultiStageQueryContext.getIndexSpec(queryContext, jsonMapper);
-
-    final boolean finalizeAggregations = MultiStageQueryContext.isFinalizeAggregations(queryContext);
+    final int rowsPerSegment = MultiStageQueryContext.getRowsPerSegment(sqlQueryContext, DEFAULT_ROWS_PER_SEGMENT);
```

Comment:
nit: Should we push these limits to MutliStageQueryContext itself ?
Then the function signature becomes `MultiStageQueryContext.getRowsPerSegment(sqlQueryContext)`

Reply:
Sounds like a good idea. Looks like we never call it with any other default. I'll change it.